test/framework/provider/wait.go

```diff
@@ -52,27 +61,35 @@ func WaitForAllProvidersInstalled(ctx context.Context, c client.Client, interval
 }
 
 // Wait for Provider to be successfully updated.
-func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration) error {
+func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration, t *testing.T) error {
 	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
 		l := &v1.ProviderRevisionList{}
 		if err := c.List(ctx, l); err != nil {
 			return false, err
 		}
+
 		// There should be a revision present for the initial revision and the upgrade.
 		if len(l.Items) != 2 {
+			t.Log("The no. of provider revisions is not equal to 2")
+			for i, item := range l.Items{
+				t.Logf("Provider revision %v : %v uses package %v", i, item.Name, item.Spec.Package)
+			}
 			return false, nil
 		}
-		for _, p := range l.Items {
+		for _, item := range l.Items {
 			// New ProviderRevision should be Active.
-			if p.Spec.Package == p2 && p.GetDesiredState() != v1.PackageRevisionActive {
+			if item.Spec.Package == p2 && item.GetDesiredState() != v1.PackageRevisionActive {
+				t.Logf("The state of new provider revision %v having version %v is %v", item.Name, item.ObjectMeta.ResourceVersion, item.Status)
```

Comment:
this `item.ObjectMeta.ResourceVersion` field is the Kubernetes API object version - it is not related to the Package version, so it isn't helpful in the logging.  See this logging statement from the [latest test run on this PR](https://github.com/crossplane/test/runs/7841241188?check_suite_focus=true):

```
    wait.go:92: The condition of provider revision provider-aws-b53fc91447be : 2791 is {[{Healthy False 2022-08-15 15:54:51 +0000 UTC UnhealthyPackageRevision }]}
```

The `ResourceVersion` here is 2791 and isn't super helpful :)

I think `item.Spec.Package` and `item.Spec.Revision` would be more useful to log here.  This is also applicable on all the other lines that we are using `item.ObjectMeta.ResourceVersion`, so make sure to update those too.

Reply:
Ohkayy! got it